### ISSUE TITLE
UserWarning: None of the inputs have requires_grad=True

### ISSUE THREAD

**Issue by zl-liu** [modelscope/ms-swift#3935]
I saw "UserWarning: None of the inputs have requires_grad=True" in the training log. When I search for this warning in the repo I see that it shows up in many issue threads, but no one seems to be primarily concerned about this warning. Could someone please explain what's going on and whether this is considered a bug? Thank you!

If it is a bug I can file a proper bug report later. Thanks again.

**Comment by Jintao-Huang** [comment]
This issue will be raised in the following situations, and it is normal, for example: the ViT part of a multimodal model (the ViT part is not trained); during the computation of the ref_model and LoRA in algorithms like DPO and other RLHF methods.

**Comment by zl-liu** [comment]
Thank you very much for your answer! @Jintao-Huang

**Comment by TBAsoul** [comment]
> This issue will be raised in the following situations, and it is normal, for example: the ViT part of a multimodal model (the ViT part is not trained); during the computation of the ref_model and LoRA in algorithms like DPO and other RLHF methods.

If this warning occurred, how can we know whether the model was being trained successfully. I'm doing some experiments on qwen2.5vl with grpo, but I find that the reward doesn't increase during training process. I'm wondering if it is caused by the gradient.

**Comment by Jintao-Huang** [comment]
You can observe the grad_norm. If it is not zero, then it is normal.
